model_transactions.go

```diff
@@ -228,67 +223,18 @@ func (m *Transaction) isExternal() bool {
 	return m.draftTransaction == nil
 }
 
-// processTxInputs will process the transaction inputs
-func (m *Transaction) processInputs(ctx context.Context) (err error) {
-	// Pre-build the options
-	opts := m.GetOptions(false)
-	client := m.Client()
-
-	var utxo *Utxo
-
-	// check whether we are spending an internal utxo
-	for index := range m.TransactionBase.parsedTx.Inputs {
-		// todo: optimize this SQL SELECT to get all utxos in one query?
-		if utxo, err = m.transactionService.getUtxo(ctx,
-			hex.EncodeToString(m.TransactionBase.parsedTx.Inputs[index].PreviousTxID()),
-			m.TransactionBase.parsedTx.Inputs[index].PreviousTxOutIndex,
-			opts...,
-		); err != nil {
-			return
-		} else if utxo != nil { // Found a UTXO record
-
-			// Is Spent?
-			if len(utxo.SpendingTxID.String) > 0 {
-				return ErrUtxoAlreadySpent
-			}
-
-			// Only if IUC is enabled (or client is nil which means its enabled by default)
-			if client == nil || client.IsIUCEnabled() {
-
-				// check whether the utxo is spent
-				isReserved := len(utxo.DraftID.String) > 0
-				matchesDraft := m.draftTransaction != nil && utxo.DraftID.String == m.draftTransaction.ID
+func (m *Transaction) updateChainInfo(txInfo *chainstate.TransactionInfo) {
+	m.BlockHash = txInfo.BlockHash
+	m.BlockHeight = uint64(txInfo.BlockHeight)
 
-				// Check whether the spending transaction was reserved by the draft transaction (in the utxo)
-				if !isReserved {
-					return ErrUtxoNotReserved
-				}
-				if !matchesDraft {
-					return ErrDraftIDMismatch
-				}
-			}
+	if txInfo.MerkleProof != nil {
+		mp := MerkleProof(*txInfo.MerkleProof)
```

Comment:
I would create  submethod here or sth to do all of MekleRoot related stuff.

Reply:
I changed it, let me know what do you think about it